app/controllers/api/v2/application_controller.rb

```diff
@@ -3,25 +3,49 @@ module V2
     class ApplicationController < ::ApplicationController
       attr_reader :current_user
 
+      before_action :check_terms_accepted
+
       def authenticate
         authenticate_or_request_with_http_token do |token|
-          if Settings.authentication.algorithm == 'plain-text'
-            # This method can be used in development mode to simplify querying
-            # the API with curl. It should allow us to do:
-            #
-            #    curl -H 'Authorization: Bearer user@education.gov.uk' http://localhost:3001/api/v2/providers
-            email = token
-          else
-            (payload, _options) = JWT.decode(token,
-                                                 Settings.authentication.secret,
-                                                 Settings.authentication.algorithm)
-            email = payload['email']
-          end
-          # match email addresses case-insensitively
+          email = email_from_token(token)
+
           @current_user = User.find_by("lower(email) = ?", email.downcase)
+
           @current_user.present?
         end
       end
+
+    private
+
+      def check_terms_accepted
+        return unless @current_user&.accept_terms_date_utc.nil?
```

Comment:
Can we use this please?

```
return if @current_user&.accept_terms_date_utc.present?
```

My mind is weak and the `unless` with a `nil?` always catches me out. :)

Reply:
I don't like `unless` either 👍